### PR TITLE
Flight API

### LINKED ISSUE #6: Flight API
The user should be able to make basic searches for flight destinations and availability based on an external flight API.

### LINKED ISSUE #17: Search Results Page

### PR DESCRIPTION
**Pull request by owal7428**
Implemented flights API and basic search page to see flights.

TODO:
- Add more search options to the search page
- Make the search page look nicer
- Integrate with calendar and weather API

NOTE:
If you are testing this page, please use POST as little as often as we only get another 90 requests for this month for the API.

Closes #6

### CODE REVIEW

**Comment by StudentKyle** [comment]
Really hope that the 90 requests per month issue doesn't hurt us later, but I'll trust your judgement here.

**Review by StudentKyle — APPROVED** [review]
Looking at how you structured it, I expect it will be very easily added into the complete search bar I made, so this looks great.

However, I can't test the code without the API key, so you should wait for more reviews before merging.

File: All project code/index.js
Comment (StudentKyle) [inline All project code/index.js:225]: Even better that this seems to be the only user input we need to take. That data will come from the search bar once the city coordinate API is complete.

File: All project code/views/partials/menu.ejs
Comment (StudentKyle) [inline All project code/views/partials/menu.ejs:18]: I think Profile has a different purpose, like displaying data on the user from the database (like, which destinatioins the user added to their account)? Maybe instead you should make an additional navbar item.

**Review by StudentKyle — CHANGES_REQUESTED** [review]

**Comment by StudentKyle** [comment]
This PR should be renamed to be more specific: This doesn't create the results page, only some results for the flight API, as well as implementing that request as a whole. Instead of closing #17, which represents a fully completed search results page with all our data and user features, you should probably replace that with #6, which actually isn't complete until this PR is merged.

**Comment by StudentKyle** [comment]
We don't want to have a hanging branch that stays unmerged until the full feature is done like you proposed in #6, because other PRs have made progress towards this other than this branch. Let's only make a PR for #17 once that is actually complete.

**Review by StudentKyle — CHANGES_REQUESTED** [review]
See above comments. While you can't easily rename the branch 17-search-results page, you can rename this pull request to something like "Implements Flight API" and add "Closes #6" instead of "Closes #17" to the PR description.
Just doing this so the PR is able to get merged and to be used in the actual development of the full search results page instead of a stale PR.

File: All project code/views/partials/menu.ejs
Comment (owal7428) [reply All project code/views/partials/menu.ejs:18]: I just moved profile, didn't delete it.

**Review by StudentKyle — APPROVED** [review]
Looks good. Next step is to integrate it into search bar in function searchQuery()

**Pull request merged**
